### PR TITLE
extract labels to it's own document and refresh

### PR DESCRIPTION
**Pull request by shiftkey**
## Overview

As part of thinking about #5994 I found myself using different labels to indicate what pull requests were about, i.e. `docs`, `infrastructure`, `tech-debt`, etc. This made me realise it was a chance to refresh our documented list of labels, because it's not been updated for a while.

## Description

[**Updated CONTRIBUTING doc**](https://github.com/desktop/desktop/blob/issue-labels/.github/CONTRIBUTING.md)

[**New 'Issues and Pull Request Labels' doc**](https://github.com/desktop/desktop/blob/issue-labels/docs/process/labels.md)

In a nutshell:

- I moved these out from the bottom of `CONTRIBUTING.md` to their own document
- I organized the document by how we use the labels (because anyone can browse [this page](https://github.com/desktop/desktop/labels), I wanted this document to show the logical groupings of labels)
- I changed the layout to make the Markdown tables 1000% more maintainable
- I refreshed the list with updated and new labels
- ~~label names are now standardized to use `-` instead of spaces (I think this only affects `help wanted` and `good first issue`)~~ reverted because GitHub The Website likes these labels to have spaces and doesn't acknowledge others
- label names are now also standardized to use all lowercase letters (aside from `macOS` which I still feel should be named to `macos` but it looks _weird_).

I avoided documenting `future-proposal` because I think we want that to go away eventually, and I'm also open to new descriptions for any of these from people who care.

TODO:

 - [x] scan rest of repository to find `help wanted` and `good first issue` links to update

### CODE REVIEW

**Comment by shiftkey** [comment]
> I think it'd be nice to link the `labels.md` from another doc so that someone doesn't have to discover it "by chance".

@nerdneha I'm gonna suggest skipping this now for a few reasons:

 - despite being in `CONTRIBUTING.md` it never really got updated to reflect changes, so what value does linking do?
 - we already call out labels that are important to new contributors in the `README`

> If you're looking for something to work on, check out the `help wanted` label.

 - aside from the groupings, most of the labels information is already there on the labels URL, so people can already see what each label is for

But when shall we link to it? I think when we want to give the rest of our `docs/process` folder some love and organization. I don't think labels doc is more important than "how we review code" or "how we triage issues" and once we figure out how to address the discoverability of those I think labels will slot in nicely alongside that.

Thoughts?

**Comment by shiftkey** [comment]
This has been quiet for almost a week and I'm going to merge this at the end of today (my time) to pay down some outstanding PRs. I'm not sure if anyone else wants to put a ✅ on this PR to confirm they're fine with the change (my last pushes made the previous ones stale).

@nerdneha feel free to either follow up on https://github.com/desktop/desktop/pull/5999#issuecomment-433525838 in your own time or someone can open a fresh issue to keep the discussion going.

File: docs/process/labels.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+
+Sometimes issues are isolated to a specific operating system. We use these
+labels to help identify these issues, so maintainers with that setup - or
+experience with that setup - can easily find them in the issue tracker.
+
+|                        | Label name | Description |
+| ---------------------- | ---------- | ----------- |
+| [:mag_right:][linux]   | `linux`    |  Issues specific to Desktop usage on Linux |
+| [:mag_right:][macOS]   | `macOS`    |  Issues specific to Desktop usage on macOS |
+| [:mag_right:][windows] | `windows`  | Issues specific Desktop usage on Windows |
```
Comment (say25) [inline docs/process/labels.md]: Spacing of the description list is a touch off but it doesn't honestly matter. `Issues` is probably redundant here

File: docs/process/labels.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+| [:mag_right:][priority-3]              | `priority-3`              | Bugs that affect small number of users and/or relatively cosmetic in nature |
+
+### External contributions
+
+We use these labels to identify work that is ideal for external contributors to
+get involved with.
+
+|                                 | Label name         |  Description |
+| ------------------------------- | ------------------ |  ----------- |
+| [:mag_right:][good first issue] | `good first issue` | Issues marked as ideal for a brand new contributor to start with |
```
Comment (say25) [inline docs/process/labels.md]: Issues is a touch redundant here

File: docs/process/labels.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+or a specific subset of users:
+
+|                             | Label name     | Description |
+| --------------------------- | -------------- | ----------- |
+| [:mag_right:][codemirror]   | `codemirror`   | Issues related to our use of [CodeMirror](https://codemirror.net/) that may require upstream fixes |
+| [:mag_right:][electron]     | `electron`     | Issues related to our use of [Electron](https://electronjs.org) that may need updates to Electron or upstream fixes |
+| [:mag_right:][integrations] | `integrations` | Issues related to editor and shell integrations that ship in Desktop |
+| [:mag_right:][performance]  | `performance`  | Relating to things affecting performance |
+| [:mag_right:][themes]       | `themes`       | Issues related the light or dark themes that ship in Desktop |
+| [:mag_right:][website]      | `website`      | Issues that relate to external websites and require co-ordination to resolve |
```
Comment (say25) [inline docs/process/labels.md]: Only two items here don't follow the `Issues related to ` format. Consistency? `Issues` is probably redundant here given the description above

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -21,9 +21,6 @@ time as the team learns, listens and refines how we work with the community.
   * [Suggesting Enhancements](#suggesting-enhancements)
   * [Help Wanted](#help-wanted)
 
-[Additional Notes](#additional-notes)
-  * [Issue and Pull Request Labels](#issue-and-pull-request-labels)
```
Comment (say25) [inline .github/CONTRIBUTING.md]: Do we want a link to `labels.md` from anywhere? maybe the root README?

Comment (shiftkey) [reply .github/CONTRIBUTING.md]: I'm 👍 to just linking to the new doc here for now

File: docs/process/labels.md (hunk 2 above)
Comment (shiftkey) [reply docs/process/labels.md]: I'm still somewhat attached to the "Issues" prefix here to differentiate them from "Issues and pull requests ..." and now that you point this out, those places that we aren't following that convention now stand out that. If someone with stronger opinions wants to have a go at the verbiage after this PR, they can be my guest.

File: docs/process/labels.md (hunk 3 above)
Comment (shiftkey) [reply docs/process/labels.md]: See my previous point ⬆️

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (shiftkey) [reply .github/CONTRIBUTING.md]: 6f61a65 has a section called Process Docs that lists some interesting things. We can refine/expand this later.

File: docs/process/labels.md (hunk 1 above)
Comment (shiftkey) [reply docs/process/labels.md]: Tidied up the whitespace in a942abb. ⬆️ has the discussion about the Issues prefixing pattern.

**Comment by shiftkey** [comment]
#jolo

**Pull request merged**